nixos/modules/services/monitoring/fluent-bit.nix

```
@@ -0,0 +1,109 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.fluent-bit;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  configurationFile =
+    if (cfg.configurationFile == null) then
+      (yamlFormat.generate "fluent-bit.yaml" cfg.configuration)
+    else
+      cfg.configurationFile;
+in
+{
+  options.services.fluent-bit = {
+    enable = lib.mkEnableOption "Fluent Bit";
+    package = lib.mkPackageOption pkgs "fluent-bit" { };
+    configurationFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        Fluent Bit configuration. See
+        <https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/yaml>
+        for supported values.
+
+        {option}`configurationFile` takes precedence over {option}`configuration`.
+
+        Note: Restricted evaluation blocks access to paths outside the Nix store.
+        This means detecting content changes for mutable paths (i.e. not input or content-addressed) can't be done.
+        As a result, `nixos-rebuild` won't reload/restart the systemd unit when mutable path contents change.
+        `systemctl restart fluent-bit.service` must be used instead.
+      '';
+      example = "/etc/fluent-bit/fluent-bit.yaml";
+    };
+    configuration = lib.mkOption {
```

Let's follow [RFC 42](https://github.com/NixOS/rfcs/pull/42) here

```suggestion
    settings = lib.mkOption {
```

---

IMO, it is more user-friendly to call it `settings`. That way, users can tell at a glance that it is a structured configuration option adhering to the `settings`-pattern.

---

option 2. seems to be the common pattern in the rest of NixOS